v2/cmd/musa/musa.go

```diff
@@ -0,0 +1,223 @@
+)
+
+type Config struct {
+	Host        string
+	Port        int
+	ProtocolID  string
+	MetricsHost string
+	MetricsPort int
+	TraceHost   string
+	TracePort   int
+}
+
+func (c Config) String() string {
+	data, _ := json.MarshalIndent(c, "", "  ")
+	return string(data)
+}
+
+var cfg = Config{
+	Host:        "127.0.0.1",
+	Port:        0,
+	ProtocolID:  string(dht.ProtocolIPFS),
+	MetricsHost: "127.0.0.1",
+	MetricsPort: 3232,
+	TraceHost:   "127.0.0.1",
+	TracePort:   14268,
+}
+
+func main() {
+	app := &cli.App{
+		Name:   "musa",
+		Usage:  "a lean bootstrapper process for any network",
+		Action: daemonAction,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "host",
+				Usage:       "the network musa should bind on",
+				Value:       cfg.Host,
+				Destination: &cfg.Host,
+				EnvVars:     []string{"MUSA_HOST"},
+			},
+			&cli.IntFlag{
+				Name:        "port",
+				Usage:       "the port on which musa should listen on",
+				Value:       cfg.Port,
+				Destination: &cfg.Port,
+				EnvVars:     []string{"MUSA_PORT"},
+				DefaultText: "random",
+			},
+			&cli.StringFlag{
+				Name:        "protocol",
+				Usage:       "the libp2p protocol for the DHT",
+				Value:       cfg.ProtocolID,
+				Destination: &cfg.ProtocolID,
+				EnvVars:     []string{"MUSA_PROTOCOL"},
+			},
+			&cli.StringFlag{
+				Name:        "metrics-host",
+				Usage:       "the network musa metrics should bind on",
+				Value:       cfg.MetricsHost,
+				Destination: &cfg.MetricsHost,
+				EnvVars:     []string{"MUSA_METRICS_HOST"},
+			},
+			&cli.IntFlag{
+				Name:        "metrics-port",
+				Usage:       "the port on which musa metrics should listen on",
+				Value:       cfg.MetricsPort,
+				Destination: &cfg.MetricsPort,
+				EnvVars:     []string{"MUSA_METRICS_PORT"},
+				DefaultText: "random",
+			},
+			&cli.StringFlag{
+				Name:        "trace-host",
+				Usage:       "the network musa trace should be pushed to",
+				Value:       cfg.TraceHost,
+				Destination: &cfg.TraceHost,
+				EnvVars:     []string{"MUSA_TRACE_HOST"},
+			},
+			&cli.IntFlag{
+				Name:        "trace-port",
+				Usage:       "the port to which musa should push traces",
+				Value:       cfg.TracePort,
+				Destination: &cfg.TracePort,
+				EnvVars:     []string{"MUSA_TRACE_PORT"},
+				DefaultText: "random",
+			},
+		},
+	}
+
+	sigs := make(chan os.Signal, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, os.Interrupt, os.Kill)
+	go func() {
+		sig := <-sigs
+		slog.Info("Received signal - Stopping...", "signal", sig.String())
+		signal.Stop(sigs)
+		cancel()
+	}()
+
+	if err := app.RunContext(ctx, os.Args); err != nil {
+		slog.Error("application error", "err", err)
+		os.Exit(1)
+	}
+}
+
+func daemonAction(cCtx *cli.Context) error {
+	slog.Info("Starting musa daemon process with configuration:")
+	fmt.Println(cfg.String())
+
+	exporter, err := prometheus.New()
```

Comment:
Check whether a metrics port/host was given and skip creating the prometheus collector or running the serve metrics goroutine if no config is supplied.

Same for tracing

Reply:
done 👍